src/Microsoft.PowerShell.ConsoleHost/host/msh/NativeHost.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static class NativeHost
```

Comment:
My post for discussion from original issue: PowerShellAssemblyLoadContext is internal :-( Utils.cs too. I suggest to start the prototype with public sealed class UnmanagedPSEntry. Alternative is to put the public class in SMA.